packages/treasury/test/swingsetTests/governance/test-governance.js

```diff
@@ -0,0 +1,103 @@
+// @ts-check
+
+// TODO Remove babel-standalone preinitialization
+// https://github.com/endojs/endo/issues/768
+import '@agoric/babel-standalone';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import rawTest from 'ava';
+import { buildVatController, buildKernelBundles } from '@agoric/swingset-vat';
+import bundleSource from '@agoric/bundle-source';
+import { E } from '@agoric/eventual-send';
+import path from 'path';
+
+import liquidateBundle from '../../../bundles/bundle-liquidateMinimum.js';
+import autoswapBundle from '../../../bundles/bundle-multipoolAutoswap.js';
+import stablecoinBundle from '../../../bundles/bundle-stablecoinMachine.js';
+import contractGovernorBundle from '../../../bundles/bundle-contractGovernor.js';
+import committeeBundle from '../../../bundles/bundle-committee.js';
+import binaryVoteCounterBundle from '../../../bundles/bundle-binaryVoteCounter.js';
+
+const filename = new URL(import.meta.url).pathname;
+const dirname = path.dirname(filename);
+
+/** @type {import('ava').TestInterface<{ data: { kernelBundles: any, config: any } }>} */
+const test = rawTest;
+
+test.before(async t => {
+  const kernelBundles = await buildKernelBundles();
+
+  const nameToBundle = [
+    ['liquidateMinimum', liquidateBundle],
+    ['autoswap', autoswapBundle],
+    ['treasury', stablecoinBundle],
+    ['committee', committeeBundle],
+    ['contractGovernor', contractGovernorBundle],
+    ['binaryVoteCounter', binaryVoteCounterBundle],
+  ];
+  const contractBundles = {};
+  nameToBundle.forEach(([name, bundle]) => {
+    contractBundles[name] = bundle;
+  });
+
+  const vatNames = ['alice', 'owner', 'priceAuthority', 'voter', 'zoe'];
+  const vatNameToSource = vatNames.map(name => {
+    const source = `${dirname}/vat-${name}.js`;
+    return [name, source];
+  });
+  const bootstrapSource = `${dirname}/bootstrap.js`;
+  vatNameToSource.push(['bootstrap', bootstrapSource]);
+
+  const bundles = await Promise.all(
+    vatNameToSource.map(([_, source]) => bundleSource(source)),
+  );
+  const vats = {};
+  [...vatNames, 'bootstrap'].forEach(
+    (name, index) => (vats[name] = { bundle: bundles[index] }),
+  );
+
+  vats.bootstrap.parameters = { contractBundles };
+
+  const config = { bootstrap: 'bootstrap', vats };
+  config.defaultManagerType = 'xs-worker';
+
+  t.context.data = { kernelBundles, config };
+});
+
+async function main(t, argv) {
+  const { kernelBundles, config } = t.context.data;
+  const controller = buildVatController(config, argv, { kernelBundles });
+  await E(controller).run();
+  return E(controller).dump();
+}
+
+const expectedTreasuryLog = [
```

Comment:
@warner @michaelfig did you tell me that this sort of "golden master" testing is frowned on now? I kinda like this style. Is that argument written down anywhere?